chainnotifier_client.go

```diff
@@ -162,6 +173,14 @@ func (s *chainNotifierClient) RegisterSpendNtfn(ctx context.Context,
 		return nil
 	}
 
+	processReorg := func() {
+		if opts.ReOrgChan == nil {
+			return
+		}
+
+		opts.ReOrgChan <- struct{}{}
```

Comment:
I think we should also listen on `ctx.Done()` here, see `RegisterConfirmationsNtfn`.

Reply:
On the same note, I think we should also do this before writing to the `errChan` (and fix conf notifier too).

Reply:
Added this in another commit.